maggy/core/reporter.py

```diff
@@ -45,21 +45,23 @@ def __init__(self, log_file, partition_id, task_attempt, print_executor):
         self.task_attempt = task_attempt
         self.print_executor = print_executor
 
+        self.env = environment_singleton()
+
         # Open executor log file descriptor
         # This log is for all maggy system related log messages
-        if not hopshdfs.exists(log_file):
-            hopshdfs.dump("", log_file)
-        self.fd = hopshdfs.open_file(log_file, flags="w")
+        if not self.env.exists(log_file):
+            self.env.dump("", log_file)
+        self.fd = self.env.open_file(log_file, flags="w+")
```

Comment:
it's a newly created file, no need to truncate it. also the log file should contain all logs from the executor, even if it's multiple trials.